api/all/src/main/java/io/opentelemetry/api/common/Value.java

```diff
@@ -115,4 +115,8 @@ static Value<List<KeyValue>> of(Map<String, Value<?>> value) {
    */
   // TODO(jack-berg): Should this be a JSON encoding?
   String asString();
+
+  static ValueBuilder builder() {
```

Comment:
this is specifically for building "maps"

another option is in #6761 that has separate map and list builders:

https://github.com/open-telemetry/opentelemetry-java/pull/6761/files#diff-11421657bbb9763b8bd1579fe3bc147d5955abc85a01a94eb567f641cdbb17a7R87-R93